doc/LanguageClient.txt

```diff
@@ -386,6 +386,124 @@ Maximum severity to show diagnostic messages.
+        \   {"Type": ['*', 'entity.name.type.class.java', '**']},
+        \ ]
+        \ }
+
+The maps associate the highlight group with a semantic scope matching pattern.
+Any symbols that have a scope that matches the pattern will be highlighted
+wtih that highlight group.
+
+There are a fixed set of semantic scopes defined by the LSP server on startup.
+These can be viewed by calling |LanguageClient_semanticScopes| which will
+show all the semantic scopes and their currently mapped highlight group for
+the currently open buffer's filetype.
+>
+     call LanguageClient_semanticScopes()
+
+     == Output from eclipse.jdt.ls ==
+
+     Highlight Group:
+      None
+     Semantic Scope:
+      invalid.deprecated.java
+       meta.class.java
+        source.java
+
+     Highlight Group:
+      None
+     Semantic Scope:
+      variable.other.autoboxing.java
+       meta.method.body.java
+        meta.method.java
+         meta.class.body.java
+          meta.class.java
+           source.java
+
+     ...
+
+Like the patterns, the semantic scope is a list of strings. They are printed
+with increasing indent to make it easier to read. For example the first scope
+is:
+>
+     ['invalid.deprecated.java', 'meta.class.java', 'source.java']
+
+It is currently isn't mapped to any highlight group as indicated by the None.
+
+Often its more useful to find what semantic scope corresponds to a piece of
+text. This can be done by calling |LanguageClient_showCursorSemanticHighlightSymbols|
+while hovering over the text of interest.
+>
+     call LanguageClient_showCursorSemanticHighlightSymbols()
+
+In order for a pattern to match a semantic scope the lists must fully match
+in length and list items.
+>
+     ['x', 'y', 'z'] == ['x', 'y', 'z']
+
+To make this easier two special strings '*' and '**' were added:
+
+Any ('*'):
+When a '*' is present it can match any SINGLE string at that position.
+>
+     ['x', '*', 'z'] == ['x', 'y', 'z']
+     ['x', '*', 'z'] != ['x', 'abc', 'y', 'z']
+
+Glob ('**'):
+When a '**' is present at the beginning and/or the end of the list it matches
+an arbitrary number of strings in its place. Currently, putting '**' in the
+middle is not implemented.
+>
+     ['x', 'y', '**'] == ['x', 'y', 'z', '1', '2', '3']
+     ['**', '123', '**'] == ['x', 'y', 'z', '123', '1', '2', '3']
+     ['**', 'y', 'z'] != ['x', 'y', 'z', '1', '2', '3']
+
+
+Example configuration for eclipse.jdt.ls:
+>
+     let g:LanguageClient_semanticHighlightMaps = {}
+     let g:LanguageClient_semanticHighlightMaps['java'] = [
+                 \ {"JavaStaticMemberFunction": ['storage.modifier.static.java', 'entity.name.function.java', '**']},
+                 \ {"JavaMemberVariable": ['meta.definition.variable.java', 'meta.class.body.java', 'meta.class.java', '**']},
+                 \ {"Function": ['entity.name.function.java', '**']},
+                 \ {"Function": ['*', 'entity.name.function.java', '**']},
```

Comment:
Could these two lines be replaced with one `{"Function": ['**', 'entity.name.function.java', '**']}`?